apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/RequestResponsePopulateMdcFilter.java

```diff
@@ -174,12 +176,18 @@ else if (headerValues.size() == 1)
       }
     }
 
+    // set to null
+    request.setAttribute(RESPONSE_DURATION_MILLISECONDS, null);
     // Record the response duration.
     Long requestStartMilliseconds = (Long) request.getAttribute(REQUEST_START_KEY);
-    if (requestStartMilliseconds != null)
+    if (requestStartMilliseconds != null) {
+      String responseDurationInMilliseconds =
+          Long.toString(System.currentTimeMillis() - requestStartMilliseconds);
       BfdMDC.put(
           BfdMDC.computeMDCKey(MDC_PREFIX, RESPONSE_PREFIX, "duration_milliseconds"),
-          Long.toString(System.currentTimeMillis() - requestStartMilliseconds));
+          responseDurationInMilliseconds);
+      request.setAttribute(RESPONSE_DURATION_MILLISECONDS, responseDurationInMilliseconds);
+    }
```

Comment:
wondering -- can we just move the calculation of response_duration into the DataServerLauncherApp so that we don't have to pass stuff back and forth using request attributes? it would be a slightly more accurate duration as well I think.